okhttp/src/main/kotlin/okhttp3/EventListener.kt

```diff
@@ -422,6 +422,34 @@ abstract class EventListener {
   ) {
   }
 
+  /**
+   * Invoked when a call fails due to cache rules. e.g. onlyIfCached and no cache entry.
+   *
+   * Events will only be received when a Cache is configured for the client.
+   */
+  open fun cacheFailure(call: Call, response: Response) {
```

Comment:
I think this wants to be called unsatisfiableRequest ? You don’t need a cache to get this error!

okhttp3.CallTest#onlyIfCachedReturns504WhenNotCached

Reply:
or maybe `satisfactionFailure` ?!